osu.Game.Tests/Visual/Online/TestSceneChatOverlay.cs

```diff
@@ -96,6 +100,58 @@ public void TestCloseChannelWhileSelectorClosed()
             AddAssert("Selector is visible", () => chatOverlay.SelectionOverlayState == Visibility.Visible);
         }
 
+        private Channel nextChannel;
+
+        [Test]
+        public void TestCloseChannelWhileActive()
+        {
+            AddUntilStep("Join until dropdown has channels", () =>
+            {
+                // Using temporary channels because they don't hide their names when not active
+                Channel toAdd = new Channel() { Name = $"test channel {JoinedChannels.Count()}", Type = ChannelType.Temporary };
+
+                ((BindableList<Channel>)channelManager.AvailableChannels).Add(toAdd);
+                channelManager.JoinChannel(toAdd);
+
+                chatOverlay.ChannelTabControl.UpdateSubTree(); // Necessary for the TabControl items to update before count check
+
+                // This test assumes only one channel being added to the dropdown
+                return VisibleChannels.Count() < JoinedChannels.Count();
+            });
+
+            AddStep("Switch to last tab", () => clickDrawable(chatOverlay.TabMap[VisibleChannels.Last()]));
+            AddAssert("Channel is last visible", () => channelManager.CurrentChannel.Value == VisibleChannels.Last());
+
+            // Closing the last channel before dropdown
+            AddStep("Close current channel", () =>
+            {
+                nextChannel = JoinedChannels.Except(VisibleChannels).First();
+                chatOverlay.ChannelTabControl.RemoveChannel(channelManager.CurrentChannel.Value);
+            });
+            AddAssert("Channel changed to next", () => channelManager.CurrentChannel.Value == nextChannel);
+
+            // Depending on the window size, one more channel needs to be closed for SelectorTab to appear
+            AddUntilStep("Close channels until selector visible", () =>
+            {
+                if (chatOverlay.ChannelTabControl.VisibleItems.Last().Name != "+")
+                {
+                    chatOverlay.ChannelTabControl.RemoveChannel(VisibleChannels.Last());
+                    chatOverlay.ChannelTabControl.UpdateSubTree();
+                    return true;
+                }
+                else return false;
+            });
+
+            // Closing the last channel with dropdown no longer present
+            AddStep("Close last when selector next", () => chatOverlay.ChannelTabControl.RemoveChannel(VisibleChannels.Last()));
+            AddAssert("Channel changed to previous", () => channelManager.CurrentChannel.Value == VisibleChannels.Last());
```

Comment:
In the previous check, I needed to explicitly save the next channel for comparison, since it's in the dropdown menu, but here I check by just assuming that the previous channel will become the last one after removing the current last channel. Is that okay or should this have more explicit checks?

Reply:
Should be fine.